src/query/weight.rs

```diff
@@ -7,12 +7,21 @@ use crate::{DocId, Score, TERMINATED};
 /// `DocSet` and push the scored documents to the collector.
 pub(crate) fn for_each_scorer<TScorer: Scorer + ?Sized>(
     scorer: &mut TScorer,
+    requires_scoring: bool,
```

Comment:
Can we not add this parameter to this function and instead:

```rust
pub(crate) fn for_each_docset<TDocSet: DocSet + ?Sized>(
    docset: &mut TDocset,
    callback: &mut dyn FnMut(DocId)
) -> { ... }
```

Reply:
I updated the PR, but it still adds a fake score when `collect()`ing